File: java/e-find2ndlargestnumber.java
class find2ndlargestnumber {
    public static void main(String[] args) {
        int[] nums = { 10, 5, 25, 8, 30, 15 };

        int first = nums[0];
        int second = nums[0];

        for (int i = 0; i < nums.length; i++) {
            if (nums[i] > first) {
                second = first;
                first = nums[i];
            }
            else if(nums[i] > second && nums[i] != first) {
                second = nums[i];
            }
        }
        System.out.println(second);
    }
}

File: python/e-find2ndlargestnumber.py
def find_2nd_largest_number(nums):
    first = nums[0]
    second = nums[0]

    for i in range(len(nums)):
        if nums[i] > first:
            second = first
            first = nums[i]
        elif nums[i] > second and nums[i] != first:
            second = nums[i]

    return second


print(find_2nd_largest_number([10, 5, 25, 8, 30, 15]))  # 25
